lib/jekyll/site.rb

```diff
@@ -2,7 +2,7 @@
 
 module Jekyll
   class Site
-    attr_reader   :source, :dest, :cache_dir, :config
+    attr_reader   :source, :dest, :cache_dir, :config, :sanitized_baseurl, :sanitized_url
```

Comment:
I'm not sure how I feel about adding new public `attr_reader`s to `Jekyll::Site`, since they'll become part of the public API. 

Are there any disadvantages to always returning a sanitized baseurl and a sanitized url?

Reply:
The disadvantage is unnecessary allocations. Technically, I'm only using `site.sanitized_baseurl`. The `sanitized_url` is just being provided for completeness.

Since you're on the fence about this, let us wait and see if the optimizations make a difference with a future state of `master`